docs/keystone-auth.md

```diff
@@ -0,0 +1,38 @@
+An authentication backend for Openstack's Keystone is maintained in this
+source tree as `haas.ext.auth.keystone`. This document describes its
+usage.
+
+# Configuration
+
+As with any other extension, you must load the extension in `haas.cfg`:
+
+    [extensions]
+    haas.ext.auth.keystone =
+
+The backend must then be configured to talk to your keystone server.
+The keystone project maintains documentation on how to do this at:
+
+<http://docs.openstack.org/developer/keystonemiddleware/middlewarearchitecture.html>
+
+Configuring HaaS to talk to Keystone deviates in the following ways:
+
+* The paste configuration is not used; you can simply ignore the
+  sections that refer to paste.
+* The options that the Keystone documentation puts in the section
+  `[keystone_authtoken]` should instead be placed in the extension's
+  section in `haas.cfg`, i.e. `[haas.ext.auth.keystone]`.
+
+# Usage
```

Comment:
Could we move this section above `Configuration`? I feel like it will likely be referred to more frequently.